### PR TITLE
Add Dockerfile.

### PR DESCRIPTION
**Pull request by cuzzo**
Hey @koistya,

First, thanks for this awesome starter kit. I tried a few others, but unlike this one, they didn't **just work**. So, thanks for that.

Although this does **just work**, I think a Dockerfile would be helpful. Not necessarily to ease the setup, but just to demonstrate how easily it can integrate into a Docker-based stack.

If you don't want unnecessary complications in the repository, I understand. Another thing that's so awesome about this starter kit is that it has everything you need to get started, without it being crazy complicated like some of the others.

Thanks,

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+RUN mkdir -p /usr/rsk
+WORKDIR /usr/rsk
+
+COPY package.json ./
+RUN npm install
+
+EXPOSE 3000
+EXPOSE 3001
+EXPOSE 5000
+VOLUME /usr/rsk
```
Comment (kevinsimper) [inline Dockerfile]: No need for defining a volume

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM mhart/alpine-node
+
+RUN mkdir -p /usr/rsk
```
Comment (kevinsimper) [inline Dockerfile]: You don't have to create it, it will be created automaticly.

Comment (cuzzo) [inline Dockerfile]: @kevinsimper, Thanks for the feedback!

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+rsk:
+  build: "."
+  ports:
+    - "0.0.0.0:3000:3000"
+    - "0.0.0.0:3001:3001"
+    - "0.0.0.0:5000:5000"
```
Comment (tobiasmuehl) [inline docker-compose.yml]: what's on port 5000?

Comment (immuzov) [inline docker-compose.yml]: @tobiasmuehl  I think it was reserved for debugger.

Comment (frenzzy) [inline docker-compose.yml]: Previously, it was the default port for the development. [RSK Changelog](https://github.com/kriasoft/react-starter-kit/blob/master/CHANGELOG.md) says:
> [v0.5.0](https://github.com/kriasoft/react-starter-kit/compare/v0.4.1...v0.5.0) (2016-02-27)
> Change the default Node.js server port from `5000` to `3000`

Currently RSK is uses `3000`, `3001` and `3002` ports by default, see [Getting Started]:(https://github.com/kriasoft/react-starter-kit/blob/master/docs/getting-started.md#3-run-npm-start)
> http://localhost:3000/ — Node.js server (build/server.js)
> http://localhost:3001/ — BrowserSync proxy with HMR, React Hot Transform
> http://localhost:3002/ — BrowserSync control panel (UI)

**Comment by kachkaev** [comment]
Any updates on this guys? Would love to see the boilerplate integrated with Docker!

**Comment by gardner** [comment]
I just added the files from this pull request into my repo. Thanks @cuzzo!!

**Comment by jshin47** [comment]
Before you add that Docker file please add the line

RUN apk add --no-cache make gcc g++ python

before RUN npm install , otherwise it fails

**Comment by jshin47** [comment]
Another change I had to make (maybe I am missing something big here)

I had to modify this line to `COPY ./ ./` 

My entire file is below. runs with `docker run imageName:tag npm run start`

```
FROM mhart/alpine-node

WORKDIR /usr/rsk

COPY ./ ./
RUN apk add --no-cache make gcc g++ python
RUN npm install

EXPOSE 3000
EXPOSE 3001
EXPOSE 5000
```

**Comment by jshin47** [comment]
I still cannot get it to work actually, but it's definitely closer. Now I am just getting an error about sqlite3. I tried to force a global install, `RUN npm i --global sqlite3`, but that didn't fix anything.

```

            ^
Error: Please install sqlite3 package manually
    at new ConnectionManager (/usr/rsk/node_modules/sequelize/lib/dialects/sqlite/connection-manager.js:25:13)
    at new SqliteDialect (/usr/rsk/node_modules/sequelize/lib/dialects/sqlite/index.js:12:28)
    at new Sequelize (/usr/rsk/node_modules/sequelize/lib/sequelize.js:233:18)
    at Object.<anonymous> (/usr/rsk/build/webpack:/data/sequelize.js:13:19)
    at __webpack_require__ (/usr/rsk/build/webpack:/webpack/bootstrap 4094f62eb0ceceaeb13a:19:1)
    at Object.module.exports.Object.defineProperty.value (/usr/rsk/build/webpack:/data/models/index.js:10:1)
    at __webpack_require__ (/usr/rsk/build/webpack:/webpack/bootstrap 4094f62eb0ceceaeb13a:19:1)
    at Object.<anonymous> (/usr/rsk/build/webpack:/core/passport.js:18:1)
    at __webpack_require__ (/usr/rsk/build/webpack:/webpack/bootstrap 4094f62eb0ceceaeb13a:19:1)
    at Object.<anonymous> (/usr/rsk/build/webpack:/server.js:25:1)
    at __webpack_require__ (/usr/rsk/build/webpack:/webpack/bootstrap 4094f62eb0ceceaeb13a:19:1)
    at /usr/rsk/build/webpack:/webpack/bootstrap 4094f62eb0ceceaeb13a:39:1
    at Object.<anonymous> (/usr/rsk/build/webpack:/webpack/bootstrap 4094f62eb0ceceaeb13a:39:1)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
/usr/rsk/tools/runServer.js:75
        throw new Error(`Server terminated unexpectedly with code: ${ code } signal: ${ signal }`);
        ^
            ^
Error: Please install sqlite3 package manually
    at new ConnectionManager (/usr/rsk/node_modules/sequelize/lib/dialects/sqlite/connection-manager.js:25:13)
    at new SqliteDialect (/usr/rsk/node_modules/sequelize/lib/dialects/sqlite/index.js:12:28)
    at new Sequelize (/usr/rsk/node_modules/sequelize/lib/sequelize.js:233:18)
    at Object.<anonymous> (/usr/rsk/build/webpack:/data/sequelize.js:13:19)
    at __webpack_require__ (/usr/rsk/build/webpack:/webpack/bootstrap 4094f62eb0ceceaeb13a:19:1)
    at Object.module.exports.Object.defineProperty.value (/usr/rsk/build/webpack:/data/models/index.js:10:1)
    at __webpack_require__ (/usr/rsk/build/webpack:/webpack/bootstrap 4094f62eb0ceceaeb13a:19:1)
    at Object.<anonymous> (/usr/rsk/build/webpack:/core/passport.js:18:1)
    at __webpack_require__ (/usr/rsk/build/webpack:/webpack/bootstrap 4094f62eb0ceceaeb13a:19:1)
    at Object.<anonymous> (/usr/rsk/build/webpack:/server.js:25:1)
    at __webpack_require__ (/usr/rsk/build/webpack:/webpack/bootstrap 4094f62eb0ceceaeb13a:19:1)
    at /usr/rsk/build/webpack:/webpack/bootstrap 4094f62eb0ceceaeb13a:39:1
    at Object.<anonymous> (/usr/rsk/build/webpack:/webpack/bootstrap 4094f62eb0ceceaeb13a:39:1)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
/usr/rsk/tools/runServer.js:75
        throw new Error(`Server terminated unexpectedly with code: ${ code } signal: ${ signal }`);
        ^

```

**Comment by jshin47** [comment]
Ok, I think I figured it out. I was copying too much - in particular my local `node_modules`.

So you need a .dockerignore file with the same contensts as .gitignore

**Comment by mjsisley** [comment]
@jshin47 

> Another change I had to make (maybe I am missing something big here)
> 
> I had to modify this line to COPY ./ ./
> 
> My entire file is below. runs with docker run imageName:tag npm run start

The PR is using a volume specified with docker-compose.
If you run `docker-compose up` it should work without `COPY ./ ./` in the dockerfile. If you want to use `docker run` then you need  `COPY ./ ./` as you did.

**Comment by jshin47** [comment]
doh... I missed that

**Comment by kmx411** [comment]
Im using this Dockerfile now, but can't seem to get it going. Maybe someone can point me in the right direction?

Here's what I did:
all in the root dir
- Add the Dockerfile
- copy .gitignore to .dockerignore
- added the docker-compose.yml file
- ran `docker-compose up` and it ran perfectly 
- when I hit localhost:3000, it stalls forever :( 
- if I run `npm start`, localhost:3000 works perfectly

**Comment by cuzzo** [comment]
@kmx411 are you on Mac? You probably have to use the IP of your Docker Machine.

**Comment by ClemensSahs** [comment]
In my mind its better to split container Repo from the App Source Code?

So rkt or Docker or something else can used.

**Comment by iDVB** [comment]
Has anyone gotten this repo to work on Docker for development? (Eg. Hot reloading etc).
I can't seem to get it to run on docker at all. I keep getting that `Please install sqlite3 package manually`.

To be honest, its not clear above what the fix is/was in the end. I've tried, the .dockerignore and removing `COPY ./ ./`

D

**Comment by kmx411** [comment]
@iDVB I couldn't get it to work for development. Instead we put the Dockerfile in the build directory and it works perfectly there.

Here's what we have:
```
FROM mhart/alpine-node

WORKDIR /usr/rsk

COPY ./ ./
RUN apk add --no-cache make gcc g++ python
RUN npm install

EXPOSE 8000
CMD [ "npm", "start" ]
```

**Comment by iDVB** [comment]
"Didn't get it working for Dev", so you use it for prod only?
Looks like the RSK team already added a working prod [Dockerfile to master](https://github.com/kriasoft/react-starter-kit/blob/master/Dockerfile). I've not tried it yet though.

**UPDATE** - Well I got everything working for Dev using `docker-compose` however if you edit files too quickly.... I end up causing this issue...  #759 

No fix yet. :(

**Comment by kmx411** [comment]
@iDVB yeah, we use it only to deploy on AWS

**Comment by koistya** [comment]
@iDVB @kmx411 the `Dockerfile` in the root of the project is intended to be used for building a production image. For running/developing the app inside Docker it might be a good idea to create `docker-compose.yml` file instead. Take a look here as an example [Node.js API Starter Kit](https://github.com/kriasoft/nodejs-api-starter) (see `docker-compose.yml`).

**Comment by ulani** [comment]
@cuzzo thank you very much for this PR! Unfortunately, we have close it due to inactivity. Feel free to re-open it or [join](https://discord.com/invite/2nKEnKq) our [Discord](https://discord.com/invite/2nKEnKq) channel for [discussion](https://github.com/kriasoft/react-starter-kit/discussions/1950).

NOTE: The `main` branch has been updated with React Starter Kit v2, using JAM-style architecture.

**Pull request closed without merging**